incuna_auth/middleware/utils.py

```diff
@@ -0,0 +1,85 @@
+from django.contrib import messages
+from django.http import HttpResponseForbidden, HttpResponseRedirect
+from django.utils.translation import ugettext_lazy as _
+
+from ..models import AccessStateExtensionMixin as AccessState
+
+
+class MiddlewareMixin:
+    def assert_request_has_user(self, request):
+        """Assert that a request has a user and return an error if it doesn't."""
+        message = (
+            "This middleware requires authentication middleware to be installed.",
+            "Edit your MIDDLEWARE_CLASSES setting to insert",
+            "'django.contrib.auth.middleware.AuthenticationMiddleware.'",
+            "If that doesn't work, ensure your TEMPLATE_CONTEXT_PROCESSORS",
+            "setting includes 'django.core.context_processors.auth'.",
+        )
+
+        assert hasattr(request, 'user'), ' '.join(message)
+
+    def get_page_access_state(self, request):
```

Comment:
I think that the feincms stuff needs to be decoupled from the base url checking here. Can it be in another class, please?